src/main/scala/com/fulcrumgenomics/sv/tools/AggregateSvPileup.scala

```diff
@@ -0,0 +1,495 @@
+    |threshold can be merged if there is a third breakpoint that is close to both.
+    |
+    |If a bam file is provided, each aggregated pileup is annotated with the allele frequency at its left and right
+    |breakends. Allele frequency is defined as the total number of templates supporting constituent breakpoints divided
+    |by the count of the union of templates that cross any constituent breakends. In particular, paired templates that
+    |straddle a breakend are considered to cross the breakend.
+    |
+    |If a bed file of target regions is provided, each aggregated pileup is annotated with whether its left and
+    |right sides overlap a target region.
+    |
+    |The output file is a tab-delimited table with one record per aggregated cluster of pileups. Aggregated
+    |pileups are reported with the minimum and maximum (inclusive) coordinates of all pileups in the cluster, a
+    |possible putative structural variant event type supported by the pileups, and the sum of read support from all
+    |pileups in the cluster.
+    |""")
+class AggregateSvPileup
+(@arg(flag='i', doc="Input text file of pileups generated by SvPileup") input: FilePath,
+ @arg(flag='b', doc="Bam file for allele frequency analysis. Must be the same file that was used as input to SvPileup.")
+ bam: Option[PathToBam] = None,
+ @arg(flag='f', doc="If bam file is provided: distance upstream and downstream of aggregated breakpoint regions to " +
+   "search for mapped templates that overlap breakends") flank: Int = 200,
+ @arg(doc="If bam file is provided: minimum total number of templates supporting an aggregated breakpoint to report " +
+   "allele frequency. Supports speed improvement by avoiding querying and iterating over huge read pileups that " +
+   "contain insufficient support for a breakpoint to be considered interesting.") minTotal: Int = 10,
+ @arg(doc="If bam file is provided: minimum allele frequency to report. Supports speed improvement by " +
+   "avoiding iterating over huge read pileups that contain insufficient support for a breakpoint to be considered " +
+   "interesting.") minFrequency: Double = 0.001,
+ @arg(flag='t', doc="Optional bed file of target regions") targetsBed: Option[FilePath] = None,
+ @arg(flag='o', doc="Output file") output: FilePath,
+ @arg(flag='d', doc="Distance threshold below which to cluster breakpoints") maxDist: Int = 10,
+) extends SvTool with LazyLogging {
+
+  Io.assertReadable(input)
+  Io.assertCanWriteFile(output)
+
+  override def execute(): Unit = {
+
+    // Read pileups from input file
+    val pileups: Seq[BreakpointPileup] = Metric.read[BreakpointPileup](input)
+    logger.info(f"Read ${pileups.length} pileups from file $input")
+
+    // Open bam reader
+    val samSource: Option[SamSource] = bam match {
+      case None => None
+      case Some(path) => Some(SamSource(path))
+    }
+
+    // Read targets from bed file and group by contig
```

Comment:
More generally I think you could load these up straight into an `OverlapDetector` (class from HTSJDK) which is basically a set of `IntervalTree`s, one per contig.  Then you can very efficiently query for overlaps.

Reply:
I think the OverlapDetector also works with anything that implements `Locatable` which includes `BEDFeature` so no need to convert to Interval either.